File: LeetCode/1742盒子中小球的最大数量.java
class Solution {
    public int countBalls(int lowLimit, int highLimit) {
        Integer[] ans = new Integer[100];
        Arrays.fill(ans, 0);
        for (int i = lowLimit; i <= highLimit; i++) {
            ans[calcNum(i)]++;
        }
        Arrays.sort(ans, new Comparator<Integer>() {
            public int compare(Integer a, Integer b) {
                return b - a;
            }
        });
        //也可以写成
        // Arrays.sort(ans, (a, b) -> (b - a));
        // Arrays.sort(ans, (a, b) -> Integer.compare(b, a));
        return ans[0];
    }

    public int calcNum(int num) {
        int ans = 0;
        while (num > 0) {
            ans += num % 10;
            num /= 10;
        }
        return ans;
    }
}

File: LeetCode/1742盒子中小球的最大数量.go
func countBalls(lowLimit int, highLimit int) int {
	nums := make([]int, 100)
	for i := lowLimit; i <= highLimit; i++ {
		nums[calcSum(i)]++
	}
	sort.Slice(nums, func(a, b int) bool {
		return nums[a] > nums[b]
	})
	return nums[0]
}

func calcSum(num int) int {
	ans := 0
	for num > 0 {
		ans += num % 10
		num /= 10
	}
	return ans
}